src/orderbook_fetcher.ts

```diff
@@ -68,18 +71,36 @@ export class OrderbookFetcher {
   }
 }
 
-function updateOrderbooks(orderbook: StreamedOrderbook) {
+function updateOrderbooks(orderbook: StreamedOrderbook)
+  : [Map<string, Orderbook>, Uint8Array] {
   const orderbooks = new Map()
+  const encodedOrders: number[] = []
   for (const item of orderbook.getOpenOrders()) {
-    addItemToOrderbooks(orderbooks, {
+    const order: Order<BN> =
+    {
       ...item,
       sellTokenBalance: new BN(item.sellTokenBalance.toString()),
       priceNumerator: new BN(item.priceNumerator.toString()),
       priceDenominator: new BN(item.priceDenominator.toString()),
       remainingAmount: new BN(item.remainingAmount.toString()),
-    })
+    }
+    addItemToOrderbooks(orderbooks, order)
+    encodeOrder(order, encodedOrders)
```

Comment:
ubernit: maybe we can make the two methods semantically similar and call it addItemToEncodedOrders and make order the second argument.